src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -1756,11 +1817,15 @@ class SQLTargetWHandle : public StorageTargetWHandle {
     if (stream_) {
       stream_.close();
     }
-    if (sqlite3_changes(db_.get()) > 0) {
-      auto statement =
-          db_.prepareStatement<std::string>("DELETE FROM target_images WHERE targetname=?;", target_.filename());
-      if (statement.step() != SQLITE_DONE) {
-        LOG_ERROR << "could not delete " << target_.filename() << " from sql storage";
+
+    if (storage_ != nullptr) {
+      SQLite3Guard db = storage_->dbConnection();
+      if (sqlite3_changes(db.get()) > 0) {
```

Comment:
There is another instance here.

Reply:
Aha, I'm so careless. Will modify it now.